src/actions/__tests__/search.test.js

```diff
@@ -28,152 +28,182 @@ const stateWithoutScopeQuery = {
-
-    const expectedActions = [
-      { type: types.SEARCH_REQUEST, payload: { q: 'test' } },
-      {
-        type: CALL_HISTORY_METHOD,
-        payload: { args: [testQueryUrl], method: 'push' },
-      },
-      { type: types.SEARCH_SUCCESS, payload: {} },
-    ];
-
-    const state = {
-      ...stateWithoutScopeQuery,
-      router: { location: { query: { page: 2 } } },
-    };
-
-    const store = getStoreWithState(state);
-    await store.dispatch(search({ q: 'test' }));
-    expect(store.getActions()).toEqual(expectedActions);
-    done();
-  });
-
-  it('creates SEARCH_SUCCESS and pushes new history state without reseting page=1 if location has page param and there is a new query with page', async done => {
-    const testQueryUrl = '/test?page=3';
-    mockHttp.onGet(testQueryUrl).replyOnce(200, {});
-
-    const expectedActions = [
-      { type: types.SEARCH_REQUEST, payload: { page: '3' } },
-      {
-        type: CALL_HISTORY_METHOD,
-        payload: { args: [testQueryUrl], method: 'push' },
-      },
-      { type: types.SEARCH_SUCCESS, payload: {} },
-    ];
-
-    const state = {
-      ...stateWithoutScopeQuery,
-      router: { location: { query: { page: 2 } } },
-    };
-
-    const store = getStoreWithState(state);
-    await store.dispatch(search({ page: '3' }));
-    expect(store.getActions()).toEqual(expectedActions);
-    done();
-  });
-
-  it('creates SEARCH_SUCCESS and pushes new history state without existing location query', async done => {
-    const testQueryUrl = '/test?filter2=value2';
-    mockHttp.onGet(testQueryUrl).replyOnce(200, {});
-
-    const expectedActions = [
-      { type: types.SEARCH_REQUEST, payload: { filter2: 'value2' } },
-      {
-        type: CALL_HISTORY_METHOD,
-        payload: { args: [testQueryUrl], method: 'push' },
-      },
-      { type: types.SEARCH_SUCCESS, payload: {} },
-    ];
-
-    const state = {
-      ...stateWithoutScopeQuery,
-      router: { location: { query: { filter1: 'value1' } } },
-    };
-
-    const store = getStoreWithState(state);
-    await store.dispatch(search({ filter2: 'value2' }, true));
-    expect(store.getActions()).toEqual(expectedActions);
-    done();
-  });
-
-  it('creates SEARCH_ERROR when search fails', async done => {
-    const testQueryUrl = '/test?size=10&q=test';
-    mockHttp.onGet(testQueryUrl).networkError();
-
-    const expectedActions = [
-      { type: types.SEARCH_REQUEST, payload: { q: 'test' } },
-      {
-        type: CALL_HISTORY_METHOD,
-        payload: { args: [testQueryUrl], method: 'push' },
-      },
-      { type: types.SEARCH_ERROR, payload: undefined },
-    ];
-
-    const store = getStoreWithState(stateWithScopeQuery);
-    await store.dispatch(search({ q: 'test' }));
-    expect(store.getActions()).toEqual(expectedActions);
-    done();
+  describe('pushQueryToLocation', () => {
+    it('pushes new location url to history', async done => {
+      const expectedUrl = '/test?size=10&q=test';
```

Comment:
shouldn't the expected be ```size=10&page=1&q=test``?

Reply:
why do you assume that? 

`page=1` is set only if a new query is made and there is already `page` param set in the current location.